flocker/node/agents/ebs.py

```diff
@@ -107,6 +115,63 @@ def _wait_for_volume(expected_volume,
             )
 
 
+def _check_blockdevice_size(device, size):
+    """
+    Helper function to check if the size of block device with given
+    suffix matches the input size.
+
+    :param unicode device: Name of the block device to check for size.
+    :param int size: Size, in SI metric bytes, of device we are interested in.
+
+    :returns: True if a block device with given name has given size.
+        False otherwise.
+    """
+    device_name = b"/dev/" + device.encode("ascii")
+
+    # Retrieve size of device as OS sees it using `lsblk`.
+    # Requires util-linux-ng package on CentOS, and
+    # util-linux on Ubuntu.
+    # Required package is installed by default
+    # on Ubuntu 14.04 and CentOS 7.
+    command = [b"/bin/lsblk", b"-nb", b"-o", b"SIZE", device_name]
+
+    # Get the base device size, which is the first line in
+    # `lsblk` output. Ignore partition sizes.
+    command_output = check_output(command).split(b'\n')[0]
```

Comment:
This could fail if `lsblk` if the given device either doesn't exist or isn't associated with an actual block device - `lsblk` exits with status 1 and `check_output` converts this to an exception.  Probably not critical that we fix this now but please file an issue and link to it from here.